### ISSUE TITLE
Exception in a continuation prompt context with `PredictionSource` as `HistoryAndPlugin` and `PredictionViewStyle` as `ListView`

### ISSUE THREAD

**Issue by lewis-yeung** [PowerShell/PSReadLine#3347]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://www.powershellgallery.com/packages/PSReadLine)
- [X] Search the existing issues, especially the pinned issues.

### Exception report

```console
### Environment
PSReadLine: 2.2.5
PowerShell: 7.2.4
OS: Microsoft Windows 10.0.22000
BufferWidth: 110
BufferHeight: 45

Last 94 Keys:

 UpArrow DownArrow DownArrow " Enter
 Enter
 Enter
 Enter
 Enter
 Enter
 Enter
 Enter
 Enter
 Enter
 Enter
 Enter
 Enter
 Enter
 Enter
 Enter
 Enter
 Enter
 Enter
 Enter
 Enter
 Enter
 Enter
 Enter
 Enter
 Enter
 Enter
 Enter
 Enter
 Enter
 Enter
 Enter
 Enter
 Enter
 Enter
 Enter
 Enter
 Enter
 Enter
 Enter
 Enter
 Enter
 Enter
 Enter
 Enter
 Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace

### Exception

System.IndexOutOfRangeException: Index was outside the bounds of the array.
   at System.Text.StringBuilder.get_Chars(Int32 index)
   at Microsoft.PowerShell.PSConsoleReadLine.ConvertOffsetToPoint(Int32 offset)
   at Microsoft.PowerShell.PSConsoleReadLine.ReallyRender(RenderData renderData, String defaultColor)
   at Microsoft.PowerShell.PSConsoleReadLine.ForceRender()
   at Microsoft.PowerShell.PSConsoleReadLine.Render()
   at Microsoft.PowerShell.PSConsoleReadLine.BackwardDeleteChar(Nullable`1 key, Object arg)
   at Microsoft.PowerShell.PSConsoleReadLine.ProcessOneKey(PSKeyInfo key, Dictionary`2 dispatchTable, Boolean ignoreIfNoAction, Object arg)
   at Microsoft.PowerShell.PSConsoleReadLine.InputLoop()
   at Microsoft.PowerShell.PSConsoleReadLine.ReadLine(Runspace runspace, EngineIntrinsics engineIntrinsics, CancellationToken cancellationToken, Nullable`1 lastRunStatus)
```

### Screenshot

![screenshot-1](https://user-images.githubusercontent.com/83903009/174291404-8f32e7e2-2b06-4dea-bde6-ed2a7365a99f.png)

![screenshot-2](https://user-images.githubusercontent.com/83903009/174291413-34687e59-34c3-4ace-bfeb-df10da09a196.png)

### Environment data

```console
PS Version: 7.2.4
PS HostName: ConsoleHost
PSReadLine Version: 2.2.5
PSReadLine EditMode: Windows
OS: 10.0.22000.1 (WinBuild.160101.0800)
BufferWidth: 136
BufferHeight: 16
```

### Steps to reproduce

1. Run PowerShell without a profile (`pwsh -noprofile`).
2. In a new prompt, run `Set-PSReadlineOption -PredictionSource 'HistoryAndPlugin' -PredictionViewStyle 'ListView'`. This is **the paramount prerequisite**.
3. Enter a single or double quote character (`'`/`"`). This line is the "initial line".
4. Press <kbd>Enter</kbd> repeatedly in a continuation prompt context until the "initial line" is out of the upper bound of the terminal screen buffer.
5. Press <kbd>Backspace</kbd> repeatedly in a continuation prompt until the "initial line" is back.
6. The above exception happens.

Something weird: The exception does not happen when there are five or more characters (including the unmatched quote) in the "initial line".

### Expected behavior

No exception.

### Actual behavior

The above exception happens.

**Comment by daxian-dbw** [comment]
> BufferWidth: 110
BufferHeight: 45

I cannot reproduce the reported issue (using the same buffer height/width, also tried other height/width). Are you able to reproduce this consistently? If so, can you be more specific on the repro steps -- such as how many `Enter` to press and how many `Backspace` to press -- to make sure it reproduceable?

![CompletionPredictor](https://user-images.githubusercontent.com/127450/174856871-ad242c11-9ee4-4308-81ab-9510ef5b3917.gif)

**Comment by lewis-yeung** [comment]
@daxian-dbw Found the culprit! There is a line `Set-PSReadlineOption -PredictionSource 'HistoryAndPlugin' -PredictionViewStyle 'ListView'` in my profile. With this prerequisite, we can reproduce the bug stably.

Seems that if both of the two features are enabled, something goes wrong internally.

**Comment by daxian-dbw** [comment]
@lewis-yeung Thanks, that helps!

**Comment by rafek1241** [comment]
Additional information:
![image](https://user-images.githubusercontent.com/9272027/177280173-f962e769-f305-42ab-bc88-234859369092.png)
**There's no PredictionViewStyle option.**
Powershell version: **7.2.5**
PSReadLine version:
![image](https://user-images.githubusercontent.com/9272027/177280477-efebfe6a-0a52-4ff7-8e5c-4bcde9935853.png)

**Comment by daxian-dbw** [comment]
@rafek1241 It's possible the help content on your machine is outdated. You can run `Get-Command Set-PSReadLineOption -Syntax` to see the latest available parameters.

**Comment by rafek1241** [comment]
@daxian-dbw 
![image](https://user-images.githubusercontent.com/9272027/177390970-e4bb23d9-0b59-4ed6-aa2f-204619fdd360.png)

**Comment by daxian-dbw** [comment]
It looks the PSReadLine version you imported in this session is 2.1.0. Run `Get-Module PSReadLine` in the same session to see if that's the case.

**Comment by rafek1241** [comment]
@daxian-dbw  Should I make a GIF? Above screens visualize same session and same terminal - I have 2.2.6 version and such command

**Comment by rafek1241** [comment]
@daxian-dbw What about my issue? Should I create another bug case in repo?

**Comment by daxian-dbw** [comment]
`PredictionViewStyle` parameter is available with PSReadLine 2.2.2+.

`Get-InstalledModule` list the modules installed from PowerShellGallery, but the PSReadLine loaded in the current session is not necessarily the same one. You should try `Get-Module PSReadLine` from the session and check the version of that.

**Comment by rafek1241** [comment]
@daxian-dbw 
You're right! My version from `Get-Module` is 2.1.0. Now I need to find location and fix it.. ;/
